pkg/virt-controller/watch/vmi.go

```diff
@@ -2447,3 +2449,52 @@ func (c *VMIController) syncMemoryHotplug(vmi *virtv1.VirtualMachineInstance) {
 		vmi.Labels[virtv1.MemoryHotplugOverheadRatioLabel] = *overheadRatio
 	}
 }
+
+func (c *VMIController) aggregateDataVolumesConditions(vmiCopy *virtv1.VirtualMachineInstance, dvs []*cdiv1.DataVolume) {
+	if len(dvs) == 0 {
+		return
+	}
+
+	dvsReadyCondition := virtv1.VirtualMachineInstanceCondition{
+		Status:  k8sv1.ConditionTrue,
+		Type:    virtv1.VirtualMachineInstanceDataVolumesReady,
+		Reason:  virtv1.VirtualMachineInstanceReasonAllDVsReady,
+		Message: "All of the VMI's DVs are bound and not running",
+	}
+
+	dvmessages := []string{}
+	for _, dv := range dvs {
+		cstatus, cmessage := getReadyConditionStatus(dv.Status.Conditions)
+		if cstatus != k8sv1.ConditionTrue {
+			dvsReadyCondition.Reason = virtv1.VirtualMachineInstanceReasonNotAllDVsReady
+			if cmessage != "" {
+				dvmessages = append(dvmessages, cmessage)
+			}
+			if cstatus == k8sv1.ConditionFalse {
+				dvsReadyCondition.Status = cstatus
+			} else if cstatus == k8sv1.ConditionUnknown && dvsReadyCondition.Status == k8sv1.ConditionTrue {
```

Comment:
nit: For simplicity I think we can get rid of this and just evaluate the condition to `True` if all of the DVs have `DataVolumeReady == True` and `False` otherwise.

Reply:
status `ConditionUnknown` is a legit status for every condition so I think we should try to honor it whenever possible, see:
https://github.com/kubernetes/api/blob/b3fafe75341fe8f5ad15cfd09f9148a8a5628852/core/v1/types.go#L2863-L2871

Reply:
The actual logic is:
- all `True` -> `True`
- at least one `False` -> `False`
- no `False` ones, but at least one `Unknown` -> `Unknown`

Reporting `False` when we are sure that no DVs is really false but at least one is `Unknown` (and potentially `True` or potentially `False`) on my opinion could be misleading.